wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -1791,6 +1807,61 @@ private CodeBlock fieldInitializer(ProtoType type, @Nullable Object value) {
     }
   }
 
+  private CodeBlock identityValue(Field field) {
+    switch (field.getEncodeMode()) {
+      case MAP:
+        return CodeBlock.of("$T.emptyMap()", Collections.class);
+      case REPEATED:
+      case PACKED:
+        return CodeBlock.of("$T.emptyList()", Collections.class);
+      case NULL_IF_ABSENT:
+        return CodeBlock.of("null");
+      case OMIT_IDENTITY:
+        ProtoType protoType = field.getType();
+        Type type = schema.getType(protoType);
+        if (protoType.equals(ProtoType.STRUCT_NULL)) {
+          return CodeBlock.of("null");
+        } else if (field.isOneOf()) {
+          return CodeBlock.of("null");
+        } else if (protoType.isScalar()) {
+          if (protoType.equals(ProtoType.BOOL)) {
```

Comment:
should this be a `Map<ProtoType, CodeBlock>` ?

Reply:
Let's do it. Kotlin would need it to, I'll follow up